### DISCUSSION TITLE
[Question] Variable sharing among different browser sessions

### DISCUSSION THREAD

**Discussion by mxav1111** [benoitc/gunicorn#3225] [Q&A] [answered]
Hi There,
If I have variable in defined in main (). Just a regular variable. Not global.
with simple syntax as ...

fletapp='My 1st Browser based Flet App'

Will this variable value be shared among completely different browser sessions (and also get overwritten) ?
Basically  I have a situation where global function (defined as global so that various .pys can access it) appears to update the value of variable which is running in another browser based session. This isn't suppose to happen. It is inadvertently updating session in another browser.

I am using gunicorn (uvicorn) and calling with the following syntax...

app.mount('/', flet.fastapi.app(main))

===

Can someone please clarify?

Thanks for your help.

**Comment by benoitc** [comment, marked as answer]
you can pass environment variable at startup time. By design nothing is shared among workers until the preload setting is used.

**Comment by mxav1111** [comment]
Thank you. I disabled the global function (although it was updating flet's button value).
